doctr/transforms/modules/pytorch.py

```diff
@@ -41,11 +47,27 @@ def forward(self, img: torch.Tensor) -> torch.Tensor:
 
             # Scale image
             img = F.resize(img, tmp_size, self.interpolation)
+            raw_shape = img.shape[-2:]
             # Pad (inverted in pytorch)
             _pad = (0, self.size[1] - img.shape[-1], 0, self.size[0] - img.shape[-2])
             if self.symmetric_pad:
                 half_pad = (math.ceil(_pad[1] / 2), math.ceil(_pad[3] / 2))
                 _pad = (half_pad[0], _pad[1] - half_pad[0], half_pad[1], _pad[3] - half_pad[1])
+
+            # In case boxes are provided, resize boxes if needed (for detection task if preserve aspect ratio)
+            if target is not None:
+                if self.preserve_aspect_ratio:
+                    # Get absolute coords
+                    if target.shape[1:] == (4,):
+                        target[:, [0, 2]] *= raw_shape[-1] / self.size[1]
+                        target[:, [1, 3]] *= raw_shape[-2] / self.size[0]
+                    elif target.shape[1:] == (4, 2):
+                        target[..., 0] *= raw_shape[-1] / self.size[1]
+                        target[..., 1] *= raw_shape[-2] / self.size[0]
```

Comment:
If the aspect ratio is preserved, there might be some shifting (padding) to be done on the target as well I'd argue

I might be missing something but it looks like it's missing the padding step

Reply:
I checked with `show-samples`, it should be OK. 
What is done here is just a changing of referential, we need to multiply the coordinate of each point by the ratio old_shape / new_shape, the old shape being the shape before padding (for instance, if you pad half of the picture, you want to divide by 2 all x and y coordinates)

Reply:
![ratio](https://user-images.githubusercontent.com/70526046/154321125-f22695bb-fd85-4fdd-80fb-cbea20dfad0d.png)

Reply:
I perfectly agree with you about getting the shape & target before padding. But in the next lines, the image is padded, but the target remains unchanged and I'm concerned about that part (again I might be missing something) :upside_down_face:

Reply:
I think it is just a linear transformation since the padding is not centered (this is why the content is not centered). You have to de-correlate the rotation and the resizing, for the right and left samples we first perform a resizing with padding to keep the aspect ratio, and then we rotate the sample, so the outputs are as expected. But as you suggests, we could remove the padding in the first resizing (this resizing is just to be sure we aren't exploding the memory), then we could rotate and finally resize and pad preserving the aspect ratio.

Reply:
For the second point, I think I should indeed add the boxes to the transformation

Reply:
Oh yeah, I see, now it makes sense!
So I assume that on the first sample, it was padded on the bottom or on the right, but the rotation made it look like it's from the top right?

Either way, I would argue the first and last samples have an issue: they are padded on 2 sides (we could have expanded the content more while preserving the aspect ratio). Is that because of the double resize? If so, we should find a better way, and I may have a suggestion, say we want to resize to (W, H):
- the first resize should resize the minimum side of that image to max(W,H), (just scaling the image so that the shortest side is equal to max(W, H)) without padding anything
- we then do a rotation while adding expand=True
- then we do a resize, this time by preserving aspect ratio (thus padding)

What do you think?

Also: should we try to make it centered though?

Reply:
yes

Reply:
This should be OK now!
aspect ratio and no symmetric padding:
![nosym](https://user-images.githubusercontent.com/70526046/154526683-5da55a02-597f-432a-ac65-88ec0a1ca6c9.png)
aspect ratio and symmetric padding:
![sym](https://user-images.githubusercontent.com/70526046/154526688-db8dddd8-9c5e-464b-b914-46d1db2be389.png)
no aspect ratio:
![noratio](https://user-images.githubusercontent.com/70526046/154526690-f89a8402-f3f6-4908-94ef-fca3f5e6e8f6.png)

Reply:
I tested all the options in both tf & pt, by default I put conservation of the aspect ratio & symmetric padding in the scripts